sdks/python/apache_beam/runners/interactive/dataproc/dataproc_cluster_manager.py

```diff
@@ -84,46 +89,71 @@ def create_cluster(self, cluster: dict) -> None:
     try:
       self._cluster_client.create_cluster(
           request={
-              'project_id': self._project_id,
-              'region': self._region,
+              'project_id': self.cluster_metadata.project_id,
+              'region': self.cluster_metadata.region,
               'cluster': cluster
           })
-      _LOGGER.info('Cluster created successfully: %s', self._cluster_name)
+      _LOGGER.info(
+          'Cluster created successfully: %s',
+          self.cluster_metadata.cluster_name)
+      self.master_url = self.get_master_url(self.cluster_metadata)
     except Exception as e:
       if e.code == 409:
-        if self._cluster_name == self.DEFAULT_NAME:
+        if self.cluster_metadata in ie.current_env(
+        ).clusters.master_urls.inverse:
```

Comment:
This can be moved to the constructor. And then, in this function, check if self.master_url exists before calling the dataproc client. Essentially, disassembling the get_master_url function, only keep the TODO part in it.

Keep the 409 execution route with the info logging only. Practically, this execution route will be a branch that never gets executed if the dataproc cluster manger is created and used correctly.

Reply:
This is a much better flow, thanks for that. I've incorporated the logic now.